### ISSUE TITLE
1.2MB 5,25 doesn't work.

### ISSUE THREAD

**Issue by VeryDarkdarkman** [skiselev/floppy_bios#11]
Hi!
Hope you are well.
I've got 4 floppy setup with 2.88 3,5", 1.2 5,25" and two 1.44s 3,5".
1st  FDC integrated with MB, 2nd - ISA multi I/O card.
Everything works perfectly except 5,25"
OS can't see anything on disk at all. 

This issue is independent of FDD - it works perfectly on internal and external FDC.
There is no difference if it's first or second drive, if it's on 1st or 2nd controller.
I've tried two different drives manufactured by different vendors and the result is the same.
If drive type is set on 1.44MB then OS can list the disk but of course can't access files due to geometry difference.

Best regards
Maciej Janicki

**Comment by skiselev** [comment]
Hi,

It most definitely should work. Can you please provide the following information:
1. Floppy BIOS configuration (e.g. use "p" command from the setup menu)
2. DOS version
3. Motherboard type and Multi I/O card type or chipset. Photos would be nice (high resolution, so I can read the ICs markings)

Thanks,
Sergey

**Comment by VeryDarkdarkman** [comment]
Hi,
Hmmm, I guess that I've found the source of the problems while replying. The issue occurs only if in system BIOS FDD type is not set to none on this particular motherboard...

BTW do you plan to make your BIOS compatible with Windows 9x?

BR,
Maciek.

**Comment by skiselev** [comment]
Yes, you should set floppy drives to "none" in system BIOS setup.
I am not sure why it is not compatible with Windows 95 :)
I fact, I don't know if Windows 95 uses BIOS for floppy. At least for HDDs, it has its own driver.

**Comment by VeryDarkdarkman** [comment]
Windows in general has its own drivers for FDDs but those drivers are using information given by BIOS. 
Windows can't detect floppies on its own. 
x86 assembler isn't my strong side so I can tell you only the outcomes but I've noticed some interesting interferences.
Generally while using Your BIOS I can't boot Windows 95 - Windows Protection Error of IOS.VXD is occuring.
While using old motherboard BIOS (which is limiting HDD capacity to ~8GB) I'm able to overcome that issue by disabling FDC driver in Windows (all 4 floppies are working perfectly).
If I'm trying to use more modern BIOS or some SCSI controller with HDD larger than 8GB then there is no way to fix IOS.VXD error.
It's not bounded to one particular motherboard. 
Ah, your BIOS and XT_IDE BIOS doesn't like each others to :)

**Comment by skiselev** [comment]
Looks like the original issue (1.2MB/5.25" doesn't work) has been resolved.
Regarding XT-IDE (XUB) BIOS extension and Multi-Floppy BIOS, they do work together, but there are some caveats. For example XUB might not enumerate drives correctly if more than 2 drives are present. Also the initialization sequence might matter... If I recall correctly Multi-Floppy BIOS should be initialized first?!
